### ISSUE TITLE
[BUG] Inaccurate results with connector snowflake when data scanned exceeds threshold in aggregate functions

### ISSUE THREAD

**Issue by cole-seph-work** [awslabs/aws-athena-query-federation#1740]
**Describe the bug**
The Snowflake connector produces inaccurate column values when more than a few GB's of data are scanned and aggregated (in our case, via the `SUM()` function) during query execution. 
This occurs with both clustered and non-clustered tables within Snowflake and the threshold appears to be ~ 1 GB of scanned data. 

**To Reproduce**
Run a query utilizing an aggregate function across more than ~1GB of scanned data. Notice that the aggregate summed values (`usage`) differ between the Snowflake connector and running that query directly within Snowflake. 
Example query: 
```SQL
select date_trunc('week', a.day) as week, sum(a.amount) as usage 
from table a
group by 1
order by week
```

If the data scanned amount is reduced, (For instance, by using a `WHERE` clause), the problem goes away and data between the snowflake connector and snowflake console is consistent.

**Expected behavior**
Snowflake connector should return accurate results. 
If it cannot return inaccurate results, connector should throw a terminating error and relay that back through the Athena console. 

**Connector Details (please complete the following information):**
 - Version: 2024.3.2
 - Name: Amazon Athena Snowflake connector

**Comment by cole-seph-work** [comment]
Closing this issue for now while we investigate more internally.

**Comment by cole-seph-work** [comment]
Appears resolved in [v2024.8.1](https://github.com/awslabs/aws-athena-query-federation/releases/tag/v2024.8.1)
